a10_octavia/controller/worker/controller_worker.py

```diff
@@ -1502,3 +1502,49 @@ def _get_flavor_data(self, flavor_id):
             flavor_data = json.loads(flavor_profile.flavor_data)
             return flavor_data
         return None
+
+    def delete_load_balancer_with_housekeeping(self, pending_lb, cascade=True):
+        """Function to delete load balancer for A10 provider using Housekeeper thread"""
+        vthunder = self._vthunder_repo.get_vthunder_from_lb(
+            db_apis.get_session(), pending_lb.id)
+        if vthunder and vthunder.compute_id is not None:
```

Comment:
maybe `vthunder.compute_id is not None` check can move inside the block (line 1520-1523) then we don't need another vthunder = self._vthunder_repo.get_vthunder_from_lb() call.

Reply:
done